ragatouille/models/colbert.py

```diff
@@ -459,6 +458,19 @@ def search(
             for doc_id in doc_ids:
                 pids.extend(self.docid_pid_map[doc_id])
 
+        base_ncells = self.searcher.config.ncells
+        base_ndocs = self.searcher.config.ndocs
+
+        if k > len(self.searcher.collection):
+            print(
+                "WARNING: k value is larger than the number of documents in the index!",
+                f"Lowering k to {len(self.searcher.collection)}...",
+            )
+            k = len(self.searcher.collection)
+        if k > (32 * self.searcher.config.ncells):
+            self.searcher.configure(ncells=k // 32 + 2)
```

Comment:
This in particular

Reply:
Thanks for pointing out... I was optimising it with tiny datasets in mind (anything smaller and for a ~800 docs collection we are already returning fewer than expected documents for k=256) but I think you're right it might be a bit unwise for larger collections 🤔 ... maybe keep this logic but only if the collection is smaller than 10000 docs?

Reply:
i.e.:

ncells = min(k//32+2, 16) for collection_size < 10000, 8 for 10000 < collection_size < 100000, 4 above that.

Reply:
(I'll merge it as is for now, but please do flag if you have any issues with this implementation... So far find that in practice most users have really small datasets (a few hundreds to thousands documents) so I'm keen to strike a good balance in that area)